tests/test_measures.py

```diff
@@ -462,6 +462,59 @@ def circuit2():
         assert np.allclose(circuit1(), circuit2(), atol=tol)
 
 
+class TestSample:
+    """Tests that samples are properly calculated."""
+
+    def test_sample_dimensions(self):
+
+        # Explicitly resetting is necessary as the internal
+        # state is set to None in __init__ and only properly
+        # initialized during reset
+        dev = qml.device("lightning.qubit", wires=2, shots=1000)
+
+        dev.apply([qml.RX(1.5708, wires=[0]), qml.RX(1.5708, wires=[1])])
+
+        dev.shots = 10
+        dev._wires_measured = {0}
+        dev._samples = dev.generate_samples()
+        s1 = dev.sample(qml.PauliZ(wires=[0]))
+        assert np.array_equal(s1.shape, (10,))
+
+        dev.reset()
+        dev.shots = 12
+        dev._wires_measured = {1}
+        dev._samples = dev.generate_samples()
+        s2 = dev.sample(qml.PauliZ(wires=[1]))
+        assert np.array_equal(s2.shape, (12,))
+
+        dev.reset()
+        dev.shots = 17
+        dev._wires_measured = {0, 1}
+        dev._samples = dev.generate_samples()
+        s3 = dev.sample(qml.PauliX(0) @ qml.PauliZ(1))
+        assert np.array_equal(s3.shape, (17,))
+
+    def test_sample_values(self, qubit_device_2_wires, tol):
+        """Tests if the samples returned by sample have
+        the correct values
+        """
+
+        # Explicitly resetting is necessary as the internal
+        # state is set to None in __init__ and only properly
+        # initialized during reset
+        dev = qml.device("lightning.qubit", wires=2, shots=1000)
+
+        dev.apply([qml.RX(1.5708, wires=[0])])
```

Comment:
I wonder if adding more tests with different param values and operations would be a good idea.

Since code coverage is complaining about several lines, this may help, especially when testing for `float` vs `double` data too.

Reply:
Sounds good, I'll add some parameters here